client/extensions/zoninator/state/data-layer/zones/index.js

```diff
@@ -60,8 +60,10 @@ export const createZone = ( { dispatch }, action ) => {
 };
 
 export const announceZoneSaved = ( dispatch, { form, siteId }, data ) => {
+	const zone = fromApi( data );
+
 	dispatch( stopSubmit( form ) );
-	dispatch( updateZone( siteId, fromApi( data ) ) );
+	dispatch( updateZone( siteId, zone.id, zone ) );
```

Comment:
Alternatively, `zone.id` could not be explicitly passed since `zone` already contains the `id`, and then the reducer could just extract it from `zone`. But I don't feel strongly about this.